### PR TITLE
[meson] add mips64 support

### PR DESCRIPTION
**Pull request by Jamlys**
Provide an injection point for custom triplet to support additional architecture when cross-compiling with meson.

### CODE REVIEW

**Comment by Neumann-A** [comment]
I think we should have an injection point from the triplet for this `VCPKG_MESON_(NATIVE|TARGET)_BUILD_CPU(_FAMILY)?` or something similar.

**Comment by Jamlys** [comment]
> I think we should have an injection point from the triplet for this `VCPKG_MESON_(NATIVE|TARGET)_BUILD_CPU(_FAMILY)?` or something similar.

Good point. Let me see how to handle it in this way would be better.

**Comment by Jamlys** [comment]
> I think we should have an injection point from the triplet for this `VCPKG_MESON_(NATIVE|TARGET)_BUILD_CPU(_FAMILY)?` or something similar.

It's reasonable as what @Neumann-A said. I have modified the code while the naming of the relavant variables might be reconsider.

**Comment by Adela0814** [comment]
Please fix the conflicts. Thanks.

**Comment by Adela0814** [comment]
Are you trying to fix the problem? you can find the full failure logs from: https://dev.azure.com/vcpkg/public/_build/results?buildId=76734&view=artifacts&pathAsName=false&type=publishedArtifacts

**Comment by Adela0814** [comment]
Pinging @Jamlys  for response. Is work still being done for this PR?

File: docs/users/triplets.md [hunk 1]
```diff
@@ -33,6 +33,22 @@ Specifies the target machine architecture.
...
+Specifies the cpu family of `VCPKG_TARGET_ARCHITECTURE_MESON_CPU`.
+
+For more information about cpu families. See https://mesonbuild.com/Reference-tables.html#cpu-families
+
+### VCPKG_TARGET_ARCHITECTURE_MESON_CPU_ENDIAN
+Specifies the endian of `VCPKG_TARGET_ARCHITECTURE` for cross compilation with meson build system. This variable is optional and the default value is `little` if it is not defined.
+
+Valid options are `big` and `little`.
+
+
```
Comment (Neumann-A) [inline docs/users/triplets.md:51]: Should probably be renamed to `VCPKG_MESON_X` instead.

**Comment by Neumann-A** [comment]
I thought a bit more about this and it is probably better to not error if build/host is unkown for vcpkg and instead let the user provide a cross/native file with the build/host_machine entries. This approach is now implemented in #26617 and has thus less injection points/variables than this here.

**Comment by Jamlys** [comment]
> I thought a bit more about this and it is probably better to not error if build/host is unkown for vcpkg and instead let the user provide a cross/native file with the build/host_machine entries. This approach is now implemented in #26617 and has thus less injection points/variables than this here.

It's a good point. I have also thought about the way to enhance the flexibility by allowing the users to provide their own meson cross/native file. Meanwhile reserve the build-in configuration to keep compatibility.

**Comment by Adela0814** [comment]
Convert this PR to draft since there is no progress. Please ping us if this PR is ready for review again.

**Comment by Adela0814** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Pull request closed without merging**